imperial_coldfront_plugin/views.py

```diff
@@ -493,3 +506,102 @@ def group_membership_extend(
         "imperial_coldfront_plugin/extend_membership.html",
         dict(form=form, group_membership=group_membership),
     )
+
+
+PROJECT_ID_PREFIX = "rdf-"
+PROJECT_ID_REGEX = re.compile(PROJECT_ID_PREFIX + r"(?P<number>\d{6})")
+
+
+def get_next_rdf_project_id():
+    """Get the next available RDF project id value."""
+    rdf_id_attribute_type = AllocationAttributeType.objects.get(name="RDF Project ID")
+    group_ids = AllocationAttribute.objects.filter(
+        allocation_attribute_type=rdf_id_attribute_type,
+    ).values_list("value", flat=True)
+    id_numbers = [
+        int(PROJECT_ID_REGEX.match(group_id).groupdict()["number"])
+        for group_id in group_ids
+    ]
+    try:
+        new_project_number = max(id_numbers) + 1
+    except ValueError:
+        # no results
+        new_project_number = 1
```

Comment:
Maybe a question of personal taste, but I'm not a big fan of using exceptions for flow control in the case of simple scenarios like this one. I'd rather do:
```suggestion
    new_project_number = max(id_numbers) + 1 if id_numbers else 1
```